### DISCUSSION TITLE
"Absolute" coordinate location support

### DISCUSSION THREAD

**Discussion by ariellellouch** [DASDAE/dascore#277] [Ideas]
The "distance" coordinate is only the first step. 
Supporting Cartesian or lat/long coordinates would be useful for sorting data, angular corrections etc. 

Functionality I can think of:
From distance + anchor points (tap test results) -> absolute locations
From absolute location to angles (important for corrections)
Apply angular corrections from such angles + source location
Etc...

**Comment by d-chambers** [comment]
Good point. We can already attach non dimension coordinates as in [this example](https://dascore.org/tutorial/patch.html#adding-coordinates), but something a little nicer, perhaps that works with a dataframe you might create from a tap-test would be good. 

Can you describe what the absolute location to angles might look like? You mean defining an arc along which the points fit? 

@aissah is going to work on this.

**Comment by ariellellouch** [comment]
Thanks @d-chambers 

I was thinking about how we do it in practice for patches (i.e., contiguous segments with no slack/spools)
We know the distance along the fiber of each channel, and tap tests that map channels to XYZ / lat-long
Then, we basically interpolate between tap test points

So, after we do that interpolation, we have the XYZ of each channel. If we take the derivative of the XYZ points (let's say along the gauge length) we can get the orientation of the fiber (i.e., azimuth mostly for surface fiber, but in the general case of topography/downhole also the dip). These angles can be useful for analyzing sensitivity or applying corrections for the different seismic phases, assuming you know the source location, and also to "mark" corners in which we expect poor data.

Hope it is better explained and let me know if I can help @aissah

**Comment by d-chambers** [comment]
This was completed with #292, please re-open if you think there is more to be done here.
